internal/gatewayapi/securitypolicy.go

```diff
@@ -422,13 +420,21 @@ func (t *Translator) translateSecurityPolicyForRoute(
 			if irListener != nil {
 				for _, r := range irListener.Routes {
 					if strings.HasPrefix(r.Name, prefix) {
-						r.Security = &ir.SecurityFeatures{
-							CORS:          cors,
-							JWT:           jwt,
-							OIDC:          oidc,
-							BasicAuth:     basicAuth,
-							ExtAuth:       extAuth,
-							Authorization: authorization,
+						if errs == nil {
+							r.Security = &ir.SecurityFeatures{
+								CORS:          cors,
+								JWT:           jwt,
+								OIDC:          oidc,
+								BasicAuth:     basicAuth,
+								ExtAuth:       extAuth,
+								Authorization: authorization,
+							}
+						} else {
+							// Return a 500 direct response to avoid unauthorized access
+							r.DirectResponse = &ir.DirectResponse{
+								Body:       ptr.To("Failed to apply SecurityPolicies"),
```

Comment:
does the error matter? should we present it?

Reply:
It could be a hint for debugging, but we can't give out details in the response as it may leak control plane configurations that users don't have permission to access.

Reply:
I saw that the existing filter translation giving out the errMsg in the body, should they also be removed?